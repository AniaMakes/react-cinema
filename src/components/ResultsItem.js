import React from 'react';
import FilmDetails from './FilmDetails';

class ResultsItem extends React.Component{
	constructor(props){
		super(props)

		this.state = {
			filmData: '',
			detailsShown: false
		}

		this.getMoreData = this.getMoreData.bind(this);
		this.checkDetailsShown = this.checkDetailsShown.bind(this);
	}

	checkDetailsShown(event) {
		if (!this.state.filmData) {
			this.getMoreData(this.props.imdbID);
		} else {
			if (this.state.detailsShown) {
					this.setState({detailsShown: false})
				}
				else{
					this.setState({detailsShown: true})
				}
		}

	}

	getMoreData(imdbid) {
		fetch('http://www.omdbapi.com/?apikey=674ff9&i=' + imdbid)
		.then(results => results.json())
		.then(data => {
			this.setState({
				filmData: data,
				detailsShown: true
			})
		})
		.catch(err => console.log(err))
		
	}

	render(){


		const {title, imdbID, year, posterURL} = this.props;

		return(
			<article className='results-item' id={imdbID}>
				<div className='results-wrapper'>
					<img className='item-poster' src={posterURL} />
					<div className='item-info-wrapper'>
						<h3 className='item-title'>{title}</h3>
						<p className='item-year'>{year}</p>
						</div>
					<button
						onClick={this.checkDetailsShown}
					> More </button> 
				</div>
				<FilmDetails detailsShown={this.state.detailsShown} data={this.state.filmData}/>
			</article>
		);
	}
}

export default ResultsItem;