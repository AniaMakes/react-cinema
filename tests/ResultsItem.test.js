import React from "react";
import ResultsItem from "../src/components/ResultsItem"
import renderer from 'react-test-renderer'

describe("Results Item", () =>{
	it("results item is rendered", ()=>{

		const mock = renderer.create(<ResultsItem
			title="Random movie"
				year="2010"
				imdbID="i237hf"
				posterURL="../img/cat-image.jpeg"
				key="i237hf" />).toJSON();

		expect(mock).toMatchSnapshot();

	})
})
