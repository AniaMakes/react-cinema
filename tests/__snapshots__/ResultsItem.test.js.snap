// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Item checks that item is rendered 1`] = `
<article
  className="results-item"
  id="i237hf"
>
  <div
    className="results-wrapper"
  >
    <img
      className="item-poster"
      src="../img/cat-image.jpeg"
    />
    <div
      className="item-info-wrapper"
    >
      <h3
        className="item-title"
      >
        Random movie
      </h3>
      <p
        className="item-year"
      >
        2010
      </p>
    </div>
    <button
      onClick={[Function]}
    >
       More 
    </button>
  </div>
  <div
    className="film-details"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p
      className="film-runtime"
    >
       
      <b>
        Runtime:
      </b>
       
       
    </p>
    <p
      className="film-genre"
    >
       
      <b>
         Genre:
      </b>
       
       
    </p>
    <p
      className="film-plot"
    >
       
    </p>
  </div>
</article>
`;

exports[`Results Item esults item is rendered 1`] = `
<article
  className="results-item"
  id="i237hf"
>
  <div
    className="results-wrapper"
  >
    <img
      className="item-poster"
      src="../img/cat-image.jpeg"
    />
    <div
      className="item-info-wrapper"
    >
      <h3
        className="item-title"
      >
        Random movie
      </h3>
      <p
        className="item-year"
      >
        2010
      </p>
    </div>
    <button
      onClick={[Function]}
    >
       More 
    </button>
  </div>
  <div
    className="hide-details"
  >
    <p
      className="film-runtime"
    >
       
      <b>
        Runtime:
      </b>
       
       
    </p>
    <p
      className="film-genre"
    >
       
      <b>
         Genre:
      </b>
       
       
    </p>
    <p
      className="film-plot"
    >
       
    </p>
  </div>
</article>
`;
